src/main/java/net/pms/io/BasicSystemUtils.java

```diff
@@ -223,6 +223,11 @@ public String parsePingLine(String line) {
 		return timeString;
 	}
 
+	@Override
+	public int getPingPacketFragments(int packetSize) {
+		return ((packetSize + 8) / 1500) + 1;
+	}
```

Comment:
Both I guess. It wasn't my choice, the code was already like this - but I can imagine that obtaining the MTU size for the network interface in question isn't trivial from Java. Also, you'd need to know the MTU size of the device on the other end and of any routers in between. It's not really viable to get it.

I don't think it make a whole lot difference to the speed estimate either. The MTU size determines how many fragments the packages that's larger than the MTU is divided into. The only "consequence" that has when measuring throughput is that there's a small overhead (20 bytes or so) per fragment because of fragment headers. Even if the MTU size is changed (which is rare AFAIK), chances are that it wouldn't be changed enough to make a real difference.

MTU size of 1500 is the maximum allowed by a standard [Ethernet frame](https://en.wikipedia.org/wiki/Ethernet_frame) as defined in [IEEE_802.3](https://en.wikipedia.org/wiki/IEEE_802.3). Some server/high-speed environmens use [jumbo frames](https://en.wikipedia.org/wiki/Jumbo_frame), but I really don't think it's relevant in this context. For jumbo frames to work, all equipment must support and be configured to use it. I doubt there's any renderer device out there where you can adjust the MTU size at all.